### ISSUE TITLE
Font color on disabled buttons should take the button background color into account

### ISSUE THREAD

**Issue by stianpr** [zurb/foundation-sites#9306]
Background on disabled buttons was recently fixed in #9022, but the font color on disabled buttons doesn't take the background color into account. If one has a lighter secondary color, the text on the button is almost invisible. Shouldn't the color on the disabled button be determined by some sort of contrast control? Maybe we can use`smart-color()` with some darkening magic?

**Test case:**
http://codepen.io/anon/pen/ALkyrE

**Comment by Silvester20** [comment]
Just wondering why you used version 6.2.1 of Foundation in your codepen and not the latest version 6.2.4?

**Comment by stianpr** [comment]
> Just wondering why you used version 6.2.1 of Foundation in your codepen and not the latest version 6.2.4?

Good question! I just forked the already defined Codepen draft that was in the issue placeholder text. Did not think to look for the version there. Sorry for that! I have updated the Codepen example in the main issue with the latest Foundation now.

**Comment by kball** [comment]
@rafibomb can you take a look at the predefined codepens and update the versions of Foundation in them?

**Comment by Owlbertz** [comment]
So address this (better later than never), I created an [updated pen](https://codepen.io/Owlbertz/pen/MQjVmK).
However, please note that setting background via CSS is not part of Foundation's responsibilites as we can only work with values inside SCSS.
What is indeed one of the frameworks responsibilities is to ensure readable and thus accessible default values. What's also in scope is a proper warning if background- and font-color have not enough contrast to meet certain requirements (whether it is AA or AAA).

As [WCAG](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) states, inactive elements have no contrast requirement, but I for myself am struggling with the low contrast on my own in some cases...

As a fix, it might be good to increase the opacity for disabled buttons to 0.5 instead of the current 0.25.

Ping @ncoden @JeremyEnglert @colin-marshall

**Comment by ncoden** [comment]
I agree with @Owlbertz. We can only control the design and accessibility of the components we generated ourselves. If a component design is changed in CSS, we can only expect the developer to also take care of the hover/active/disabled states design and accessibility by himself.

About the text color of a disabled button, I would like to add 2 things:
* Having the text color of a disabled button more difficult to read is intended. This is a common way to explicitely indicate that there is no possible interactivity.
* If the text color of a disabled button is not readable by an healthy person, there is high chances for disabled people to not being able to read the non-disabled button. So you should not choose these background/text colors in the first place.

We _may_ incread the disabled button opacity to `0.33` (I find that `0.5` is way too high), but I personally don't think this is really necessary.

**Comment by Owlbertz** [comment]
Maybe we should move away from the opacity solution in general, as this has side effects with the background color. Imo we could easily mimic the current design using SASS functions (is there a way to reduce saturation or do similar color manipulation while keeping the initial base color?) and avoid opacity to indicate the disabled state.

**Comment by ncoden** [comment]
> as this has side effects with the background color

Yes, and it's what I expect. To be less visible, so darker on a dark background and lighter on a light background.

**Comment by Owlbertz** [comment]
Ok, but disabled !== less visible.

**Comment by ncoden** [comment]
Yes, but this is according to me the best way to indicate "no interactivity".

**Comment by Owlbertz** [comment]
Ok, maybe we can get some more opinions involved.

**Comment by colin-marshall** [comment]
I am no accessibility expert, so take what I say with a grain of a salt.

I personally think the combination of a 0.5 opacity with `cursor: not-allowed` gets the point across. I do see how it is kind of difficult to read due to contrast issues. What if we made it so that on hover, the disabled buttons go to a higher opacity? You still have the not-allowed cursor to let you know it's disabled. Or does having the color change on hover go against it being an element with no interactivity?

**Comment by ncoden** [comment]
> Or does having the color change on hover go against it being an element with no interactivity?

I think it does yes.

**Comment by DanielRuf** [comment]
> Ok, maybe we can get some more opinions involved.

@SassNinja what do you think?

darken / lighten / combine may make sense here.
In our own projects we use the color function of Sass.

**Comment by JeremyEnglert** [comment]
We can't control colors once a user starts making changes. That's up to them.

However, we have a lot of colors that are problematic in the default color palette. I'll work on this and the disabled buttons this weekend.

**Comment by DanielRuf** [comment]
@JeremyEnglert any updates on your side?
